packages/nimble-components/src/table/tests/table.spec.ts

```diff
@@ -161,48 +161,83 @@ describe('Table', () => {
                 moreStringData: 'foo'
             }
         ];
-        element.data = updatedData;
+        element.setData(updatedData);
         await waitForUpdatesAsync();
 
-        verifyRenderedData();
+        verifyRenderedData(updatedData);
     });
 
     it('updating data can remove rows from the table', async () => {
         await connect();
 
-        element.data = [...simpleTableData];
+        element.setData(simpleTableData);
         await waitForUpdatesAsync();
 
         const updatedData: SimpleTableRecord[] = [
             simpleTableData[0],
             simpleTableData[2]
         ];
-        element.data = updatedData;
+        element.setData(updatedData);
         await waitForUpdatesAsync();
 
-        verifyRenderedData();
+        verifyRenderedData(updatedData);
     });
 
     it('updating data can reorder rows from the table', async () => {
         await connect();
 
-        element.data = [...simpleTableData];
+        element.setData(simpleTableData);
         await waitForUpdatesAsync();
 
         const updatedData: SimpleTableRecord[] = [
             simpleTableData[1],
             simpleTableData[2],
             simpleTableData[0]
         ];
-        element.data = updatedData;
+        element.setData(updatedData);
         await waitForUpdatesAsync();
 
-        verifyRenderedData();
+        verifyRenderedData(updatedData);
+    });
+
```

Comment:
Could you add a test that calls `setData()` with an empty array? There's some conditional code in the implementation so it would be good to verify what happens if it's not called.